src/components/PetitionLinksField/index.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import domOnlyProps from 'form/domOnlyProps';
+import fieldIsInvalid from 'form/fieldIsInvalid';
+import getLinkInputErrors from 'form/getLinkInputErrors';
+import wrapPetitionLinks from 'helpers/wrapPetitionLinks';
+import styles from './petition-links-field.scss';
+
+const PetitionLinksField = React.createClass({
+  getClassname (element, error) {
+    return [
+      styles[element || 'input'],
+      styles[error ? 'invalid' : 'valid']
+    ].join(' ');
+  },
+
+  getInitialState: () => ({ value: '' }),
+
+  componentWillMount () {
+    const { helper } = this.props;
+    let links = helper.value;
+
+    links = wrapPetitionLinks(links);
+    helper.onChange(links);
+  },
+
+  handleChange (e) { this.setState({ value: e.target.value }); },
+
+  handleLinkAdded (e) {
+    const { helper, config } = this.props;
+    helper.touched = true;
+    helper.onBlur();
+
+    if (e.key === 'Enter' || e.type === 'blur') {
+      // Disable form submitting when pressing ENTER
+      e.preventDefault();
+
+      const { value } = this.state;
+      let links = helper.value;
+
+      if (value) {
+        // Specific validation for the link field.
+        const error = getLinkInputErrors(value, links, config);
+
+        if (error) {
+          helper.error = error;
+          return;
+        }
+
+        helper.error = false;
+
+        links.push({ url: value });
+        helper.onChange(links); // Update redux-form value
+        this.setState({ value: '' }); // Clear input value
+
+        // Fetch open graph data for the last link
+        this.props.fetchOpenGraph(value).then(({ openGraph }) => {
+          const newValue = { url: value, og: openGraph };
+          const lastLinkIndex = links.length - 1;
+
+          links[lastLinkIndex] = newValue;
+          helper.onChange(links);
+        });
+      }
+    }
+  },
+
+  handleLinkRemoved (index) {
+    let links = this.props.helper.value;
+
+    links.splice(index, 1);
+    this.props.helper.onChange(links);
+  },
```

Comment:
Do we not need to dispatch an action here to remove the object from the `openGraph` state object?